### ISSUE TITLE
Dynamic Variables menu

### ISSUE THREAD

**Issue by arthur-jacquin** [numworks/epsilon#1752]
#### Problem you'd like to fix

Users can use variables anywhere : functions, calculation, sequences, ... This is a great feature and it works well. However, it is needed to get back to the Calculation app to change variables values.

#### Describe the solution you'd like

A more capable Variables menu would be a perfect answer to this (minor) usability problem. Currently, the menu only displays and make accessible (by clicking EXE) saved variables, but it could let the user directly modify their values. Another idea : when a new variable name is detected but no value has been stored, the variable is automatically created in the menu with an "undef" value.

#### Additional context

I think the benefits of a more powerful menu are widespread, but here is an example :

1. The user creates a new function. A sum is involved, and as the function is quite complicated the user do not know how many iterations the calculator can handle smoothly (another case : there is constant meant to be changed a lot through experimentation). So the user type "n" in the function.
2. The user opens the Variables menu. The "n" variable, with an "undef" value, has just been automatically created. The user enters 2, for example.
3. The user goes to the Graph tab to see how fast the function is traced. It works well.
4. Without leaving the Graph tab, the user reaches for the Variables menu and try another value.

![screenshot](https://user-images.githubusercontent.com/53219173/105524833-7fb61080-5ce0-11eb-8f2b-624942c2fda8.png)
![screenshot(1)](https://user-images.githubusercontent.com/53219173/105524856-880e4b80-5ce0-11eb-977c-dfea3039f207.png)
![screenshot(2)](https://user-images.githubusercontent.com/53219173/105524865-89d80f00-5ce0-11eb-9f68-dda346126e17.png)
![screenshot(3)](https://user-images.githubusercontent.com/53219173/105524874-8ba1d280-5ce0-11eb-8cba-f2aa829390eb.png)
![screenshot(4)](https://user-images.githubusercontent.com/53219173/105524887-8fcdf000-5ce0-11eb-98ab-0cc4b8022285.png)

**Comment by artaxxx** [comment]
Thanks @arthur-jacquin for your issue, this is interesting!
I think there's a lot of cases to think about. For instance, how can the user edit a `u(n+1)` sequence but it's a really nice feature to add!

**Comment by arthur-jacquin** [comment]
You are right, functions and sequences are harder to edit on the fly. Two cases of alterations :

- either the function (or sequence) depend on a constant meant to be changed, in which case the user can edit the constant in the Expressions sub-menu of the Variables menu.
- either this is a big/structural change and I wouldn't mind changing app.

So to me the big deal is about being able to change Expressions 😄. Changing functions and sequences could easily get clumsy.

Expressions could be modified the same way it is possible to modify the *x* value in the Functions Graph tab : just start typing (with the cursor hovering the according expression) !
